crates/engine/tree/src/tree/root.rs

```diff
@@ -97,11 +97,36 @@ impl<Factory> StateRootConfig<Factory> {
     }
 }
 
+/// A proof target, this can be either:
+/// * An address, for fetching account proofs, or
+/// * An address and storage slot, for fetching storage proofs
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum ProofTarget {
+    /// Account proof target
+    Account(Address),
+    /// Storage proof target
+    Storage {
+        /// The account
+        account: Address,
+        /// The slot
+        slot: B256,
+    },
+}
+
+impl ProofTarget {
+    /// Returns the address for the account referenced in the proof target.
+    pub const fn address(&self) -> &Address {
+        match self {
+            Self::Account(account) | Self::Storage { account, .. } => account,
+        }
+    }
+}
+
 /// Messages used internally by the state root task
 #[derive(Debug)]
 pub enum StateRootMessage<BPF: BlindedProviderFactory> {
     /// Prefetch proof targets
-    PrefetchProofs(HashSet<Address>),
+    PrefetchProofs(HashSet<ProofTarget>),
```

Comment:
this can be `MultiProofTargets` cause we'll need to update the account anyway

Reply:
renamed

Reply:
I think @rkrasiuk meant using `MultiProofTargets` struct here, because we already have it ready

Reply:
oh whoops lol